### ISSUE TITLE
pyramidkv/squeeze: per-layer-varying kept-token COUNT is incompatible with mlx_lm's single shared mask-per-forward-pass, blocking the #370 mask fix

### ISSUE THREAD

**Issue by rajveer43** [rajveer43/VeloxQuant-MLX#390]
## Summary

While implementing the #370 fix (eviction-family caches silently corrupt prefill attention via `mask="causal"`), found that `pyramidkv` and `squeeze` cannot use the same fix as the other 6 eviction caches (snapkv, streaming_llm, h2o, tova, chunkkv, cam), because of a deeper architectural conflict with `mlx_lm` itself.

## The conflict

`mlx_lm` builds **one** attention mask per model forward pass, from a single representative cache (`cache[0]`), and reuses that exact same mask array unchanged for every layer's attention call:

```python
# mlx_lm/models/qwen2.py (representative of essentially every model — 202 call
# sites across mlx_lm/models/*.py follow this pattern)
mask = create_attention_mask(h, cache[0])
for layer, c in zip(self.layers, cache):
    h = layer(h, mask, c)   # same `mask` object passed to every layer
```

and inside each layer:

```python
output = scaled_dot_product_attention(queries, keys, values, cache=cache, scale=self.scale, mask=mask)
```

`mx.fast.scaled_dot_product_attention` requires `mask` to be broadcast-compatible with `[B, N, T_q, T_kv]` — where `T_kv` is **that layer's own** key count. This works today only because every layer in the non-buggy methods keeps the *same number* of tokens (a uniform, config-level budget), even though *which* tokens differ layer to layer.

`pyramidkv` and `squeeze` break this assumption by design: each layer gets its own **resolved budget**, deliberately different per layer (that's the entire point of both methods — a data/position-driven per-layer memory allocation). Confirmed empirically on `mlx-community/Qwen2.5-0.5B-Instruct-4bit`, 24 layers, `pyramid_budget=40`:

```
layer budgets: [77, 74, 71, 67, 64, 61, 58, 54, 51, 48, 45, 42, 38, 35, 32, 29, 26, 22, 19, 16, 13, 9, 6, 3]
```

Once a prefill's token count exceeds a given layer's budget, that layer's `n_kept` differs from other layers' `n_kept`. A single explicit mask built from `cache[0]`'s kept positions/count would be **shape-wrong** for every other layer with a different resolved budget — it can't just be "the same fix, applied here too."

## Why this needs its own fix, not #370's

The #370 fix (see that issue and its PR) works by building an explicit position-based causal mask from `cache[0]`'s per-(batch,head) absolute kept-token positions. That's sufficient for snapkv/streaming_llm/h2o/tova/chunkkv/cam because every layer in those methods keeps the same *count* of tokens (though not necessarily the same *specific* positions) — so a mask array shaped for `cache[0]`'s `T_kv` is shape-compatible with every other layer's `T_kv` too, and position-correct since it's built from true absolute positions rather than "whatever layer 0 specifically kept."

pyramidkv/squeeze need a structurally different resolution, e.g. (not yet evaluated, needs its own design pass):
1. **Pad every layer to a common kept-count** (e.g. `max(budget_by_layer)`), masking out padding rows as always-invisible via the same explicit-mask mechanism — but this partially defeats the point of per-layer budgeting (every layer's memory cost regresses toward the largest layer's budget, not its own smaller one), and touches the budget-resolution code in `veloxquant_mlx/cache/base.py`'s `_build_pyramidkv`/`_build_squeeze`.
2. Some other mechanism entirely (needs research) — possibly per-layer mask construction if `mlx_lm` exposes a hook for it (not confirmed to exist), or accepting a documented limitation for these two specific methods.

## Scope

- `veloxquant_mlx/cache/pyramidkv_cache.py` — `PyramidKVCache`
- `veloxquant_mlx/cache/squeeze_cache.py` — `SqueezeAttentionCache`
- `veloxquant_mlx/cache/base.py` — `KVCacheBuilder._build_pyramidkv`, `_build_squeeze` (per-layer budget resolution)

Both remain affected by the original #370 corruption (`mask="causal"` fast path) until this is resolved — they are explicitly **not** covered by #370's fix, which is scoped to the 6 uniform-budget eviction caches only.
